demo/app.js

```diff
@@ -7,15 +7,18 @@ demoApp.controller('demoCtrl', function ($scope, zfaModal) {
 
     $scope.idArray = ['myModal','alert','confirm','prompt'];
 
+    $scope.promiseResult = "";
+
     $scope.showModal = function(id) {
 
         switch(id) {
             case 'myModal':
                 zfaModal.open('myModal',{
                     template: '<div zf-modal=""><a class="close-button" zf-close="">X</a><h4>Hello!</h4>' +
-                    '<a ng-click="resolve()" class="button primary">OK</a><a ng-click="reject()" class="button secondary">Cancel</a></div>'
+                    '<a ng-click="resolve()" class="button primary">OK</a><a ng-click="cancel()" class="button secondary">Cancel</a></div>',
+                    controller: 'demoCtrl'
```

Comment:
why 'demoCtrl' here? Anw, this PR has been opened for too long, so I will merge and update it.